Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -327,12 +369,12 @@ void CGPathMoveToPoint(CGMutablePathRef path, const CGAffineTransform* m, float
 
     if (pathObj->_count + 1 >= pathObj->_max) {
```

Comment:
If this is moved to a utility function for adding a new CGPathElement, could we also put the dynamic allocation for the points array (always of the same size) in this utility function and forgo the CGPathElementInternal derived struct? This logic occurs every time a new element is created. Let me know if you had something different in mind for CGPathElementInternal, but as I understand you just wanted it to allocate memory enough memory for the max number of CGPoints and assign it to the points pointer in CGPathElement in the constructor of CGPathElementInternal.